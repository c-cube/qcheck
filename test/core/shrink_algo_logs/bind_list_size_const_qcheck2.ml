open QCheck2

let bind_pair_list_size =
  Test.make ~name:"bind list_size constant"
    Gen.(int_bound 1000 >>= fun len ->
         list_size (return len) (int_bound 1000) >>= fun xs -> return (len,xs))
    (Log.shrinks Print.(pair int (list int)) @@
     fun (len,xs) -> let len' = List.length xs in len=len' && len' < 4)
;;
try Test.check_exn bind_pair_list_size with Test.Test_fail _ -> ()
