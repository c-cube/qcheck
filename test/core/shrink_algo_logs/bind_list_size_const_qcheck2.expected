fails (385, [666; 863; 235; 895; 235; 241; 408; 927; 32; 2; 606; 896; 513; 762; 910; 973; 86; 599; 574; 612; 0; 267; 567; 196; 495; 239; 353; 864; 5; 627; 100; 121; 910; 438; 880; 536; 212; 843; 281; 323; 93; 868; 505; 649; 719; 905; 344; 708; 209; 351; 633; 22; 634; 525; 691; 556; 953; 805; 0; 844; 43; 696; 810; 707; 906; 907; 392; 2; 35; 166; 620; 377; 392; 212; 993; 144; 625; 42; 685; 926; 300; 711; 32; 297; 591; 571; 457; 187; 538; 110; 152; 339; 826; 279; 876; 720; 457; 463; 648; 718; 427; 302; 184; 281; 216; 90; 989; 505; 927; 262; 98; 905; 431; 68; 705; 843; 864; 518; 969; 186; 596; 97; 362; 493; 743; 309; 357; 275; 989; 68; 168; 256; 288; 420; 831; 399; 281; 110; 746; 887; 713; 92; 344; 345; 644; 218; 960; 515; 534; 743; 507; 329; 485; 754; 553; 172; 177; 758; 583; 809; 789; 26; 929; 183; 551; 159; 731; 92; 424; 671; 883; 500; 978; 387; 847; 264; 618; 751; 944; 573; 116; 606; 692; 312; 280; 141; 498; 107; 996; 682; 807; 331; 665; 74; 991; 361; 133; 954; 632; 403; 583; 346; 556; 963; 632; 43; 624; 318; 491; 406; 827; 76; 923; 904; 168; 587; 528; 867; 192; 337; 176; 766; 619; 167; 78; 694; 124; 21; 737; 944; 784; 480; 260; 543; 872; 30; 681; 962; 138; 848; 776; 622; 916; 680; 684; 46; 712; 510; 141; 313; 236; 120; 357; 686; 610; 277; 72; 22; 244; 754; 341; 659; 314; 661; 812; 337; 167; 384; 388; 119; 899; 25; 891; 51; 96; 49; 554; 922; 691; 976; 774; 116; 770; 117; 787; 133; 298; 745; 899; 530; 865; 448; 747; 146; 110; 749; 898; 17; 577; 620; 652; 920; 972; 169; 544; 259; 517; 420; 49; 475; 123; 331; 307; 866; 865; 889; 72; 166; 549; 76; 445; 296; 814; 769; 469; 106; 234; 715; 504; 12; 680; 298; 878; 212; 24; 611; 951; 68; 420; 915; 58; 156; 203; 967; 217; 728; 641; 981; 705; 161; 671; 322; 520; 727; 930; 512; 25; 930; 305; 96; 441; 430; 166; 269; 569; 967; 120; 347; 644; 140; 254; 241; 622; 567; 422; 283; 788; 249; 15; 124; 90; 311; 413; 572; 720])
holds (0, [])
fails (192, [324; 48; 802; 152; 131; 676; 701; 230; 810; 1000; 386; 862; 732; 433; 130; 732; 56; 711; 290; 513; 876; 630; 627; 717; 357; 218; 639; 884; 101; 916; 819; 178; 826; 185; 865; 482; 482; 405; 646; 854; 501; 783; 200; 292; 75; 56; 782; 460; 556; 909; 923; 148; 946; 542; 328; 151; 225; 749; 279; 793; 194; 316; 730; 958; 666; 770; 665; 586; 494; 205; 950; 730; 136; 368; 736; 728; 828; 230; 376; 342; 11; 48; 752; 480; 715; 499; 602; 866; 683; 230; 822; 33; 53; 606; 915; 918; 649; 0; 722; 329; 395; 694; 259; 375; 58; 4; 562; 461; 825; 994; 734; 492; 60; 181; 957; 325; 262; 112; 217; 926; 115; 836; 8; 307; 957; 421; 466; 752; 302; 733; 331; 372; 397; 528; 772; 666; 990; 572; 672; 390; 236; 491; 817; 887; 126; 161; 30; 921; 824; 648; 471; 79; 706; 2; 994; 36; 439; 901; 744; 563; 994; 980; 957; 438; 762; 254; 692; 337; 48; 799; 297; 488; 420; 737; 460; 198; 135; 169; 321; 918; 127; 722; 172; 911; 903; 527; 593; 968; 897; 645; 981; 799])
holds (0, [])
fails (96, [811; 159; 609; 97; 342; 910; 648; 616; 783; 925; 233; 650; 265; 280; 591; 464; 599; 437; 706; 547; 820; 403; 745; 888; 597; 370; 445; 833; 468; 717; 497; 174; 344; 704; 593; 65; 0; 914; 119; 862; 366; 720; 688; 814; 389; 487; 889; 402; 11; 243; 389; 722; 196; 530; 190; 808; 955; 70; 642; 284; 69; 734; 502; 924; 690; 682; 118; 434; 31; 245; 582; 341; 416; 632; 171; 248; 351; 213; 830; 781; 419; 537; 329; 345; 797; 772; 676; 393; 11; 913; 129; 558; 20; 874; 445; 150])
holds (0, [])
fails (48, [50; 103; 751; 87; 880; 449; 349; 320; 934; 701; 881; 172; 401; 202; 945; 662; 399; 397; 355; 56; 933; 812; 623; 149; 647; 989; 484; 680; 714; 90; 197; 569; 324; 821; 991; 231; 359; 217; 731; 475; 114; 458; 478; 833; 377; 685; 87; 510])
holds (0, [])
fails (24, [892; 259; 520; 697; 393; 629; 277; 683; 134; 515; 702; 836; 893; 311; 360; 712; 661; 460; 158; 99; 766; 51; 2; 838])
holds (0, [])
fails (12, [144; 562; 738; 466; 813; 679; 89; 225; 779; 595; 523; 555])
holds (0, [])
fails (6, [366; 301; 374; 889; 681; 511])
holds (0, [])
holds (3, [431; 22; 143])
fails (5, [919; 632; 872; 51; 69])
holds (0, [])
holds (2, [795; 105])
holds (3, [786; 189; 131])
fails (4, [411; 517; 387; 222])
holds (0, [])
holds (2, [77; 154])
holds (3, [183; 300; 969])
fails (4, [0; 517; 387; 222])
fails (4, [0; 0; 387; 222])
fails (4, [0; 0; 0; 222])
fails (4, [0; 0; 0; 0])
